File: tps/TP01/q3_tp1/main.c
#include <stdio.h>
#include <string.h>

void ler(char *s){
	//%d%*c
	scanf("%[^\n]%*c",s);
}
/*
void teste(char c){
	printf("%c\n",c+1);
}
*/
void cript(char *s, int chave, char *encrip){
	int tam = strlen(s);
	int i;
	for(i=0;i<tam;i++){
		encrip[i] = s[i] + chave;	
	}	
	encrip[i] = '\0';
}

void print(char *s){
	int tam = strlen(s);
	for(int i=0;i<tam;i++){
		printf("%c ",s[i]);
	}
}

int main(){
	int n = 10;
	int chave = 3;
	char str[n];
	int tam;	

	ler(str);
	tam = strlen(str) + 1;
	char encrip[tam+1];//para incluir o '\0'

	cript(str,chave,encrip);
	print(encrip);
	
	return 0;
}

//e: str
//s: str cif

File: tps/TP01/q3_tp1/Main.java
public class Main {

    public static String criptografar(String str, int chave) {

        String strCodificada;
        int tam = str.length();
        int i;

        StringBuilder aux = new StringBuilder(str);//para acessar com comportamento de vetor de char
        StringBuilder encrip = new StringBuilder();

        for (i = 0; i < tam; i++) {//construir a nova str
           char c =  aux.charAt(i);
           char novoC = (char)(c + chave);
           encrip.append(novoC);
        }
        
        strCodificada = encrip.toString();//converte novamente para str
        return strCodificada;
        
    }

    public static void main(String[] args) {

        int chave = 3;
        String str;

        do{
            str = MyIO.readLine();//leitura
            if(str.compareTo("FIM")!=0){//evita printar depois de FIM
                MyIO.println(criptografar(str, chave));
            }
        }while(str.compareTo("FIM")!=0);

    }
}